src/test/basics/DetectCrash_test.cpp

```diff
@@ -0,0 +1,44 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <beast/unit_test/suite.hpp>
+
+namespace ripple {
+namespace test {
+
+struct DetectCrash_test : public beast::unit_test::suite
+{
+    void testDetectCrash ()
+    {
+        testcase ("Detect Crash");
+        // Crash the process. This is used to test that the multi-process
+        // unit test will correctly report the crash.
+        *reinterpret_cast<volatile int*>(0) = 0;
```

Comment:
Apparently this throws an actual exception in Windows builds:
```
$ ./build/cmake/msvc/Release/rippled --unittest=DetectCrash --unittest-jobs=6   
0> beast.unit_test.DetectCrash Detect Crash
0> #1 failed: unhandled exception
5.7s, 1 suite, 1 case, 1 test total, 1 failure

$ ./build/cmake/msvc/Debug/rippled --unittest=DetectCrash --unittest-jobs=6
0> beast.unit_test.DetectCrash Detect Crash
0> #1 failed: unhandled exception
9.8s, 1 suite, 1 case, 1 test total, 1 failure

$ ./build/cmake/msvc/Debug/rippled --unittest=DetectCrash
beast.unit_test.DetectCrash Detect Crash
#1 failed: unhandled exception
142ms, 1 suite, 1 case, 1 test total, 1 failure
```

In Linux, I see the expected "may have crashed message".
```
$ ./build/cmake/gcc.debug/rippled --unittest=DetectCrash --unittest-jobs=6
0> beast.unit_test.DetectCrash Detect Crash
602ms, 0 suites, 0 cases, 0 tests total, 0 failures

Suite: beast.unit_test.DetectCrash failed to complete. The child process may have crashed.
```

Changing this to a "simple" `std::terminate()`, however, seems to work better in both environments, though Windows displays an Abort/Retry/Ignore dialog.
```
$ ./build/cmake/msvc/Debug/rippled --unittest=DetectCrash --unittest-jobs=6
1> beast.unit_test.DetectCrash Detect Crash
1.6s, 0 suites, 0 cases, 0 tests total, 0 failures

Suite: beast.unit_test.DetectCrash failed to complete. The child process may have crashed.
```

https://github.com/ximinez/rippled/commit/8a7ae1e471b38e124a17913e261c988ab1159160

Reply:
Thanks for finding that. I took your fix and pushed. I'm a little curious as to what kind of exception was thrown. Next time I boot up windows I might run a little experiment.